lib/showoff.rb

```diff
@@ -478,6 +479,178 @@ def process_content_for_replacements(content)
+        },
+        "callout" => {
+          :matchers => [
+            /^\.(?=callout)(?<classes>.*)\s+(?<inner>.*?)$/
+          ],
+          :block => proc { |match_data|
+            classes = match_data[:classes].gsub('.',' ')
+            inner = process_custom_syntax(match_data[:inner])
+            %Q(\n<div markdown="1" class="callout #{classes}">\n\n#{inner}\n\n</div>\n\n)
+          }
+        },
+        "div" => {
+          :matchers => [
+            /\.(?<classes>[^\[\!\ ]*)\[(?<indent_level>\d)\s(?<inner>.*?)\k<indent_level>\]/m,
+            /\.(?<classes>[^\[\!\ ]*)\[\D(?<inner>.*?)\D\]/m,
+            /~~~DIV:(?<classes>[^~]*)~~~(?<inner>.*?)~~~ENDDIV~~~/m
+          ],
+          :block => proc { |match_data|
+            inner = process_custom_syntax(match_data[:inner])
+            classes = match_data[:classes].gsub('.',' ')
+            %Q(\n<div markdown="1" class="#{classes}">\n\n#{inner}\n\n</div>\n\n)
+          },
+        },
+        "section" => {
+          :matchers => [
+            /~~~SECTION:(?<classes>[^~]*)~~~(?<inner>.*?)~~~ENDSECTION~~~/m
+          ],
+          :block => proc { |match_data|
+            inner = process_custom_syntax(match_data[:inner])
+            classes = match_data[:classes].gsub('.',' ')
+            %Q(\n<div markdown="1" class="notes-section #{classes}">\n\n#{inner}\n\n</div>\n\n)
+          },
+        },
+        "code" => {
+          :matchers => [
+            /\s{4}@@@\s(?<languages>.*?)\n\s{4}(?<text>.*?)\n\n/m
+          ],
+          :block => proc { |match_data|
+            css = match_data[:languages].split.collect {|l| "language-#{l.downcase}" }.join(' ')
+            text = match_data[:text].gsub(/^\s{4}/,'')
+            %Q(<pre class="highlight"><code class="#{css}">#{text}</code></pre>)
+          }
+        },
+        "paragraph" => {
+          :matchers => [
+            /^\s+\.(?!break|comment)(?<classes>\S*) (?<inner>.+)$/
+          ],
+          :block => proc { |match_data|
+            inner = process_custom_syntax(match_data[:inner])
+            classes = match_data[:classes].gsub('.', ' ')
+            %Q(\n<p markdown="1" class="#{classes}">\n\n#{inner}\n\n</p>\n\n)
+          },
+        },
+        "pagebreak" => {
+          :matchers => [
+            /~~~PAGEBREAK~~~/
+          ],
+          :block => proc {
+            %Q(<div class="pagebreak">continued...</div>)
+          }
+        },
+        "form" => {
+          :matchers => [
+            /~~~FORM:(?<title>[^~]*)~~~/
+          ],
+          :block => proc { |match_data|
+            %Q(<div class="form wrapper" title="#{match_data[:title]}"></div>)
+          }
+        },
+        # Check for any kind of options
+        "config" => {
+          :matchers => [
+            /~~~CONFIG:(?<setting>.*?)~~~/
+          ],
+          :block => proc { |match_data|
+            settings.showoff_config[match_data[:setting]] if settings.showoff_config.key?(match_data[:setting])
```

Comment:
backwards, it should replace `~~~CONFIG:foo~~~` with the the value of `settings.showoff_config['foo']`